discovery-provider/src/tasks/users.py

```diff
@@ -600,7 +600,7 @@ def update_user_events(
 ) -> None:
     """Updates the user events table"""
     try:
-        if not isinstance(events, dict):
+        if not isinstance(events, dict) or not user_record.blocknumber:
```

Comment:
why add this check?

Reply:
This shouldn't change anything - it's because the blocknumber is not nullable in the DB model and the python types think that it is optional, which it never is in the poa indexing flow.